app/bundles/LeadBundle/EventListener/SearchSubscriber.php

```diff
@@ -125,4 +139,256 @@ public function onBuildCommandList(MauticEvents\CommandListEvent $event)
             );
         }
     }
+
+    /**
+     * @param LeadBuildSearchEvent $event
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function onBuildSearchCommands(LeadBuildSearchEvent $event)
+    {
+        switch ($event->getCommand()) {
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_read'):
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_read', [], null, 'en_US'):
+                    $this->buildEmailReadQuery($event);
+                break;
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_sent'):
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_sent', [], null, 'en_US'):
+                    $this->buildEmailSentQuery($event);
+                break;
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_queued'):
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_queued', [], null, 'en_US'):
+                    $this->buildEmailQueuedQuery($event);
+                break;
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_pending'):
+            case $this->translator->trans('mautic.lead.lead.searchcommand.email_pending', [], null, 'en_US'):
+                    $this->buildEmailPendingQuery($event);
+                break;
+            case $this->translator->trans('mautic.lead.lead.searchcommand.sms_sent'):
+            case $this->translator->trans('mautic.lead.lead.searchcommand.sms_sent', [], null, 'en_US'):
+                    $this->buildSmsSentQuery($event);
+                break;
+            case $this->translator->trans('mautic.lead.lead.searchcommand.web_sent'):
+            case $this->translator->trans('mautic.lead.lead.searchcommand.web_sent', [], null, 'en_US'):
+                    $this->buildWebSentQuery($event);
+                break;
+            case $this->translator->trans('mautic.lead.lead.searchcommand.mobile_sent'):
+            case $this->translator->trans('mautic.lead.lead.searchcommand.mobile_sent', [], null, 'en_US'):
+                    $this->buildMobileSentQuery($event);
+                break;
+        }
+    }
+
+    /**
+     * @param LeadBuildSearchEvent $event
+     */
+    private function buildEmailPendingQuery(LeadBuildSearchEvent $event)
+    {
+        $q = $event->getQueryBuilder();
+        /** @var EmailRepository $emailRepo */
+        $emailRepo = $event->getEntityManager()->getRepository('MauticEmailBundle:Email');
+        $emailId   = (int) $event->getString();
+        /** @var Email $email */
+        $email = $emailRepo->getEntity($emailId);
+        if ($email instanceof Email) {
```

Comment:
`Email repository` has to return an instance of `Email`. What is the purpose of this condition?